applications/minotari_merge_mining_proxy/src/proxy/inner.rs

```diff
@@ -112,39 +112,53 @@ impl InnerService {
             ));
         }
 
-        let mut base_node_client = self.base_node_client.clone();
-        trace!(target: LOG_TARGET, "Successful connection to base node GRPC");
+        let monero_height = json["height"].as_u64().unwrap_or_default();
+        let monero_hash: Vec<u8> = Hex::from_hex(json["hash"].as_str().unwrap_or_default()).unwrap_or_default();
+        let mut base_node_height = 0;
+        let mut base_node_hash = vec![];
+        let mut p2pool_height = 0;
+        let mut p2pool_hash = vec![];
+
+        if let Some(mut p2pool_client) = self.p2pool_client.clone() {
+            let p2pool_resp = p2pool_client.get_tip_info(GetTipInfoRequest {}).await?;
+            let res = p2pool_resp.into_inner();
+
+            base_node_height = res.node_height;
+            base_node_hash = res.node_tip_hash.clone();
+            p2pool_height = res.p2pool_rx_height;
+            p2pool_hash = res.p2pool_rx_tip_hash.clone();
+        } else {
+            let mut base_node_client = self.base_node_client.clone();
+            trace!(target: LOG_TARGET, "Successful connection to base node GRPC");
 
-        let result =
-            base_node_client
-                .get_tip_info(tari_rpc::Empty {})
-                .await
-                .map_err(|err| MmProxyError::GrpcRequestError {
+            let result = base_node_client.get_tip_info(tari_rpc::Empty {}).await.map_err(|err| {
+                MmProxyError::GrpcRequestError {
                     status: err,
                     details: "get_tip_info failed".to_string(),
-                })?;
-        let height = result
-            .get_ref()
-            .metadata
-            .as_ref()
-            .map(|meta| meta.best_block_height)
-            .ok_or(MmProxyError::GrpcResponseMissingField("base node metadata"))?;
-        if result.get_ref().initial_sync_achieved != self.initial_sync_achieved.load(Ordering::SeqCst) {
-            self.initial_sync_achieved
-                .store(result.get_ref().initial_sync_achieved, Ordering::SeqCst);
-            debug!(
-                target: LOG_TARGET,
-                "Minotari base node initial sync status change to {}",
-                result.get_ref().initial_sync_achieved
-            );
+                }
+            })?;
         }
-
         info!(
             target: LOG_TARGET,
-            "Monero height = #{}, Minotari base node height = #{}", json["height"], height
+            "Monero height = #{}, Minotari base node height = #{}, P2pool height: #{}", monero_height, base_node_height, p2pool_height
         );
 
-        json["height"] = json!(cmp::max(json["height"].as_i64().unwrap_or_default(), height as i64));
+        // Add them together. You could multiply them a factor, but xmrig will generally just check if the height or the
+        // hash is different.
+        let reported_height = p2pool_height + monero_height + base_node_height;
```

Comment:
Let's move the info log to below the `reported_height` calculation and include it in the log as well.